tests/tests/test_bson.py

```diff
@@ -88,3 +88,96 @@ def test_read_bson(
                 assert len(row) == 5
                 assert row["beatle_name"] in beatles
                 assert beatles.index(row["beatle_name"]) == row["beatle_idx"] - 1
+
+
+def test_read_bson(
```

Comment:
from looking at the implementation, i'd expect that we'd be able to write a test that has duplicate fields and assert that the value is the first one, but not the second.

Reply:
it's not possible to use to python bson library to build these. I wrote a little go script and generated a file that I committed. 

I'm not sure if we should commit/run it, but it's there to prove the point.